### ISSUE TITLE
Calendar view: optimize initial load time

### ISSUE THREAD

**Issue by pdrj** [toggl/mobileapp#6541]
Apart from the main view, which is slow due to (amongst other things 😄) loading other dependencies, calendar view is the slowest one to load.

We need to figure out how to make it at least four times as fast. 'Four times' is not arbitrary - more info in the [OKR page](https://www.notion.so/toggl/2019-Q4-OKRs-1671c3057a5f49f7b54d090c560f4192?p=7d2e61a160ec4d7c95b10faeebe14913).

To be clear, we're talking about the initial load time here. Use peter/measure branch to help with measurements.

**Comment by daividssilverio** [comment]
Hey @pdrj , is this considering the new calendar VMs and fragment after the calendar project? It has now been merged into develop.

**Comment by pdrj** [comment]
Yes, I'm afraid that's the case @daividssilverio. I retook the measurements today from [this branch](https://github.com/toggl/mobileapp/commits/peter/measure), which has @sofokli1's merge commit.

**Comment by heytherewill** [comment]
I've profiled the shared portion of this. We are not doing anything wrong like duplicating expensive calls or spamming the UI with too much data. The exception being line 73 of `CalendarFragment.cs`, but that's not enough to cause jank. The problem is basically the rendering.

Merging #5946 (which depends on #6083 and #4423) might be all we need to optimize this enough. I'm removing my assignment for the time being, since @daividssilverio is much more qualified to optimize the rendering aspects of this view.

**Comment by daividssilverio** [comment]
Hello there @heytherewill and @pdrj o/

Thanks for the nice utilities from your branch @pdrj , they are quite helpful, although I'm not completely sure about their accuracy.

So, as far as I could investigate and tweak with the calendar view today I found:
- The base layout is loading fast enough 👍 
- Each page of the calendar also loads fast enough 👍 
- The problem is that  3 or 4 pages are being loaded during the initialization of the calendar, they add up and slow down the whole thing

I've added another `viewTreeObserver` for each page to see how each of them were loading. They don't add up to the calendar fragment total load time reported as measured using the nice `measureLayoutWith` method. I'm hoping that's happening because the pages are asynchronously (not sure how the view pager works in that aspect).

I've managed to reduce the times by making everything that's being done in the `CalendarDayViewPageFragment:OnViewCreated` run wrapped in a `Task.Run(SetupBindings).ConfigureAwait(false)`.

What also helped was to set the calendar day pager's adapter to have initially only 1 element and after resuming the calendar fragment, setting it to the actual size (14 days). That spread the layout inflations and code that runs when the pages are loaded a bit more.

☝️ Doing that I managed to get the `CalendarFragment` to load roughly twice as fast (at least what's reported by the `measureLayoutWith`) when compared to the `MainFragment`

For reference, in a single instance of my tests runs:
```
MainFragment layout: 933
CalendarFragment layout: 427
EditTimeEntryActivity layout: 289
```

For reference, in a single instance of the unmodified `peter/measure` branch:
```
MainFragment layout: 927
CalendarFragment layout: 969
EditTimeEntryActivity layout: 281
```

Getting that view to load 4x faster is gonna be a challenge. I'm not sure what do past what I've done in my tests, and they are quite buggy btw and wouldn't work out without proper placeholders.

Any suggestions? Ideas?

**Comment by pdrj** [comment]
Thanks for the report @daividssilverio. I agree that the method of measuring might not be perfect, but I've not been able to come up with something better - happy to hear your thoughts. That being said, I think it does give a good (even if rough) idea of how the view is performing.

As for your optimizations - I  say we take what we can. 50% is a huge improvement already and we can work from there. 👍

**Comment by daividssilverio** [comment]
@pdrj Let's keep using that method :) It's something already o/

I'll submit a PR today or tomorrow with the improvements.
